packages/react/src/Section/Section.tsx

```diff
@@ -0,0 +1,175 @@
+import React, {forwardRef, PropsWithChildren, useCallback, useMemo} from 'react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/section/section.css'
+import styles from './Section.module.css'
+
+export const SectionPaddingVariants = ['none', 'condensed', 'normal', 'spacious'] as const
+export const SectionBackgroundColors = ['default', 'inset', 'subtle', 'overlay'] as const
+
+const defaultSectionPadding = SectionPaddingVariants[2]
+
+type ResponsiveMap<T> = {
+  narrow?: T
+  regular?: T
+  wide?: T
+}
+
+type AnyString = string & NonNullable<unknown>
+type BackgroundColors = (typeof SectionBackgroundColors)[number] | AnyString // The AnyString union type is a workaround to provide autocompletion for the SectionBackgroundColors values
+type PaddingVariants = (typeof SectionPaddingVariants)[number]
+type ResponsiveBackgroundColorMap = ResponsiveMap<BackgroundColors>
+type ResponsiveBackgroundImagePositionMap = ResponsiveMap<string | string[]>
+type ResponsiveBackgroundImageSizeMap = ResponsiveMap<string | string[]>
+type ResponsiveBackgroundImageSrcMap = ResponsiveMap<string | string[]>
+type ResponsivePaddingVariantsMap = ResponsiveMap<PaddingVariants>
+
+type SectionProps = {
+  /**
+   * The HTML element used to render the section.
+   */
+  as?: 'section' | 'div'
```

Comment:
I didn't see a test case for this, but would be good to add one that verifies this default is section given how important this is

Reply:
Good catch!